libpromises/cf3parse_logic.h

```diff
@@ -1028,4 +1030,58 @@ static inline void ParserBeginBundleBody()
     P.useargs = NULL;
 }
 
+static inline void ParserHandleQuotedListItems()
+{
+    ParserDebug("\tP:%s:%s:%s:%s list append: "
+                "qstring = %s\n",
+                ParserBlockString(P.block), P.blocktype, P.blockid,
+                (P.currentclasses ?
+                    P.currentclasses : "any"),
+                P.currentstring);
+
+    if(StringContains(P.lval, "list") && *P.lval != 's')  // deal with numbers in ilist/rlist
+    {
+        char ret[PRINTSIZE(intmax_t)] = {0};
+        double value;
+
+        bool ok = DoubleFromString(P.currentstring, &value);
+        if(!ok)
+        {
+            // TODO: we dont really want DoubleFromStrings error output...
+            ParseError("Lval '%s' expects numbers", P.lval);
+            goto cleanup;
+        }
+
+        if(*P.lval == 'i')  // ilist
+        {
+
+            // TODO: figure out proper warning/error flow
+            if(StringContainsChar(P.currentstring, '.'))
+            {
+                ParseWarning(PARSER_WARNING_TRUNCATED,
+                             "Real number '%s' will be truncated",
+                             P.currentid);
+            }
+
+            xsnprintf(ret, PRINTSIZE(intmax_t), "%jd", (intmax_t)value);
+        }
+        else  // rlist
+        {
+            xsnprintf(ret, PRINTSIZE(intmax_t), "%lf", value);
+        }
+
+        RlistAppendScalar((Rlist **) &P.currentRlist,
+                          ret);
+    }
+    else
+    {
+        RlistAppendScalar((Rlist **) &P.currentRlist,
+                          (void *) P.currentstring);
+    }
```

Comment:
This is essentially to handle the `slist` case, right?

Handling this much earlier, with an early return, would be better IMO. Then all the code after it can assume it is not slist, and doesn't have to be indented inside an if.

Something like:

```
if (StringEqual(P.lval, "slist"))
```